velox/functions/prestosql/JsonFunctions.cpp

```diff
@@ -63,10 +62,75 @@ class JsonFormatFunction : public exec::VectorFunction {
   }
 };
 
+class JsonParseFunction : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VectorPtr localResult;
+
+    // Input can be constant or flat.
+    // Todo: Replace Json Validation function (folly::jsonParse) for json_parse.
+    // Current implementation of json_parse function is computationally heavy.
+    // folly::parseJson is a computationally heavy function which returns a
+    // dynamic. json_parse udf only validates a string is in json format, so it
+    // does not require such computations and the return object.
+    assert(args.size() > 0);
+    const auto& arg = args[0];
+    if (arg->isConstantEncoding()) {
+      auto value = arg->as<ConstantVector<StringView>>()->valueAt(0);
+      try {
+        folly::parseJson(value);
+      } catch (const std::exception& e) {
+        context.setError(value.size(), std::current_exception());
```

Comment:
The input to setError is the row number where the error occurred. Hence, this should be `context.setErrors(rows, ...)`